tests/test_implied_volatility.rs

```diff
@@ -2,12 +2,12 @@ mod tests {
     use blackscholes::{ImpliedVolatility, Inputs, OptionType, Pricing};
 
     // Tolerance is a bit higher due to IV being an approximation
-    const TOLERANCE: f64 = 0.0001;
+    const TOLERANCE: f64 = 1e-10;
```

Comment:
1e-10 i think it is too much on these level.
Tbh result shouldnt be with that high tolerance in these place (prediciton vs calculation).

Reply:
That's fine, though I think it should be lower tol than 1e-4. Would 1e-8 be acceptable to you?

Reply:
i will check in new black implementation, after your merge :)